File: solutions/elixir/bob/1/lib/bob.ex
defmodule Bob do
  @spec hey(String.t()) :: String.t()
  def hey(input) do
    cond do
      String.match?(input, ~r/^\s*$/)                 -> "Fine. Be that way!"
      String.match?(input, ~r/[[:upper:]].*\?\s*$/)
        and String.upcase(input) == input             -> "Calm down, I know what I'm doing!" 
      String.match?(input, ~r/[[:upper:]]/)
        and String.upcase(input) == input             -> "Whoa, chill out!" 
      String.match?(input, ~r/\?\s*$/)                -> "Sure."
      true                                            -> "Whatever."
    end
  end
end

File: solutions/javascript/bob/1/bob.js

const isYelling = (msg) => (msg == msg.toUpperCase() && msg.match(/[A-Z]/));
const isAQuestion = (msg) => msg.trimEnd().endsWith('?');
const doesnotSayAnything = (msg) => msg.trim().length == 0;
  
export function hey(message) {
  
  if (doesnotSayAnything(message)) return 'Fine. Be that way!';
  if (isAQuestion(message) && isYelling(message)) return "Calm down, I know what I'm doing!";
  if (isYelling(message)) return 'Whoa, chill out!';
  if (isAQuestion(message)) return  'Sure.';
  return 'Whatever.';
};
